xkcd/pkg/index/index.go

```diff
@@ -0,0 +1,158 @@
+package index
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/vespian/go-exercises/xkcd/pkg/pbuff"
+	"github.com/vespian/go-exercises/xkcd/pkg/types"
+	"github.com/vespian/go-exercises/xkcd/pkg/web"
+)
+
+func serialize(a types.AllStories, t types.OndiskSerialization,
+) (
+	blob []byte,
+	err error,
+) {
+	switch t {
+	case types.JSON:
+		blob, err = json.Marshal(a)
+		if err != nil {
+			return nil, fmt.Errorf("Index JSON marshaling failed: %s", err)
+		}
+	case types.Protobuf:
+		pbuffDigestableStructs := pbuff.PBAllStoriesFromAllStories(a)
+		blob, err = proto.Marshal(pbuffDigestableStructs)
+		if err != nil {
+			log.Fatal("pbuff marshaling error: ", err)
+		}
+	default:
+		panic("Unsupported serializing method")
+	}
+
+	return
```

Comment:
Truth to be told, I am not sure about them as well :| I agree they make the functions a bit obscure.

Fixing it now.